connect/transforms/src/test/java/org/apache/kafka/connect/transforms/InsertHeaderTest.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.After;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class InsertHeaderTest {
+    private InsertHeader<SourceRecord> xform = new InsertHeader<>();
+
+    @After
+    public void teardown() {
+        xform.close();
+    }
+
+    @Test(expected = ConfigException.class)
+    public void insertHeaderWithBlankName() {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "");
+        props.put("literal.value", "dummy value");
+
+        xform.configure(props);
+    }
+
+    @Test
+    public void insertHeaderWithNullValue() {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "AAA");
+        props.put("literal.value", null);
```

Comment:
How about removing this line, since the default is null. It's not possible to define a property with a null value in a Java properties file, which is why the default was chosen to be a null value. This test should verify that when the `literal.value` is not provided, a null header value is used.